src/content/en/updates/2018/12/wakelock.md

```diff
@@ -100,78 +96,99 @@ For example, a screen wake lock implies that the app should continue running.
+which *currently* is limited to just `'screen'`. In addition,
+you also need a way to abort the wake lock,
+which works through the generic `AbortController` interface.
+Therefore, you first create a new
+[`AbortController`](https://developer.mozilla.org/en-US/docs/Web/API/AbortController),
+and then pass the controller’s
+[`AbortSignal`](https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal)
+as the second parameter to `WakeLock.request()`.
+Two things can happen next that you need to `catch`:
+
+- The wake lock can after a while just be regularly aborted,
+  which you detect by checking if the exception’s name is `'AbortError'`.
+  In this context, `AbortError` is actually not an error in the common sense,
+  but just the way `AbortController` works.
+- The browser can also refuse the request for different reasons, for example,
+  because the battery charge level is too low.
+  In this case, the exception’s message will contain more details.
 
 ```js
-let wakeLockObj;
-if ('getWakeLock' in navigator) {
-  try {
-    // Create a wake lock for the type we want.
-    wakeLockObj = await navigator.getWakeLock('screen');
-    console.log('👍', 'getWakeLock', wakeLockObj);
-  } catch (ex) {
-    console.error('👎', 'getWakeLock', err);
-  }
+const wakeLockCheckbox = document.querySelector('#wakeLockCheckbox');
+
+if ('WakeLock' in window) {
+  let wakeLock = null;
+
+  const requestWakeLock = () => {
+    const controller = new AbortController();
+    const signal = controller.signal;
+    window.WakeLock.request('screen', {signal})
+    .catch((e) => {
+      if (e.name === 'AbortError') {
+        wakeLockCheckbox.checked = false;
+        console.log('Wake Lock was aborted');
+      } else {
+        console.error(`${e.name}, ${e.message}`);
+      }
+    });
+    wakeLockCheckbox.checked = true;
+    console.log('Wake Lock is active');
+    return controller;
+  };
+
+  wakeLockCheckbox.addEventListener('change', () => {
+    if (wakeLockCheckbox.checked) {
+      wakeLock = requestWakeLock();
+    } else {
+      wakeLock.abort();
+      wakeLock = null;
+    }
+  });
 }
 ```
 
-In some instances, the browser may fail to create the wake lock object, and
-instead throws an error, for example if the batteries on the device are low.
-
-
-### Use the wake lock object {: #use-wake-lock-obj }
-
-We can use the newly created `wakeLockObj` to activate a lock, or determine
-the current wake lock state and to receive notifications when the wake lock
-state is changed.
-
-To acquire a wake lock, we simply need to call `wakeLockObj.createRequest()`,
-which creates a new `WakeLockRequest` object. The `WakeLockRequest` object
-allows multiple components on the same page to request their own locks. The
-`WakeLock` object automatically handles the requests. The wake lock is
-released when all of the requests have been cancelled.
-
+### The wake lock lifecycle {: #wake-lock-lifecycle }
+
+When you play with the  [wake lock demo][demo], you will notice that
```

Comment:
Change "you will" to "you'll".

The web.dev style guide prefers contractions. Please check this throughout.

Reply:
Done